bin/check-cloudwatch-composite-metric.rb

```diff
@@ -132,7 +132,56 @@ def metric_desc
     "#{config[:namespace]}-#{config[:numerator_metric_name]}/#{config[:denominator_metric_name]}(#{dimension_string})"
   end
 
+  def composite_check
+    numerator_metric_resp = get_metric(config[:numerator_metric_name])
+    denominator_metric_resp = get_metric(config[:denominator_metric_name])
+
+    ## If the numerator is empty, then we see if there is a default. If there is a default
+    ## then we will pretend the numerator _isnt_ empty. That is
+    ## if empty but there is no default this will be true. If it is empty and there is a default
+    ## this will be false (i.e. there is data, following standard of dealing in the negative here)
+    no_num_data = resp_has_no_data(numerator_metric_resp, config[:statistics]) ? config[:numerator_default].nil? : resp_has_no_data(numerator_metric_resp, config[:statistics])
```

Comment:
I had to scroll several times just to read this and I personally prefer readability over shortcuts. Ternary operators are good for some things but usually they are relatively short any longer and I prefer to break them out:
```ruby
no_num_data = if resp_has_no_data(numerator_metric_resp, config[:statistics])
                config[:numerator_default].nil?
              else
                resp_has_no_data(numerator_metric_resp, config[:statistics])
              end
```

Reply:
Another option would be to do the assignment above with a shorter var name and then use that.